internal/cli/root/builder.go

```diff
@@ -96,3 +111,55 @@ func formattedVersion() string {
 		runtime.GOARCH,
 		runtime.Compiler)
 }
+
+func (opts *BuilderOpts) hasNewVersionAvailable() (newVersionAvailable bool, newVersion string, err error) {
+	if version.Version == "" {
+		return false, "", nil
+	}
+
+	svCurrentVersion, err := semver.NewVersion(version.Version)
+	if err != nil {
+		return false, "", err
+	}
+
+	latestVersion, err := opts.store.LatestVersion()
+	if err != nil {
+		return false, "", err
+	}
+
+	svLatestVersion, err := semver.NewVersion(latestVersion)
+	if err != nil {
+		return false, "", err
+	}
+
+	if svCurrentVersion.Compare(svLatestVersion) < 0 {
+		return true, latestVersion, nil
+	}
+
+	return false, "", nil
+}
+
+func (opts *BuilderOpts) printNewVersionAvailable(w io.Writer) error {
+	if config.SkipUpdateCheck() {
```

Comment:
not sure if people may agree but I have the opinion this should only check is a TTY, no point printing this to a file, check https://github.com/mongodb/mongocli/blob/4a13a3615cb8e6737f44d6e1108eacce753e9d5e/internal/cli/output_opts.go#L77

Reply:
@fmenezes ping on this, not a blocker, just making sure you saw it and your thoughts

Reply:
We've talked about this offline